README.md

```diff
@@ -8,6 +8,265 @@
+we display the "Enabled" lable, otherwise we consider the button to be
+"Disabled".
+
+### Enabled / Disabled Inputs
+
+Not only toggle buttons can be toggled, but also checkboxes. A checkbox is
+either enabled or disabled.
+
+Implementing a form containing a checkbox (e.g. for one of those legal
+disclaimers) is trivial:
+
+![Order chocolate](./assets/order_chocolate.png)
+
+Similar to the toggle button above, the checkbox would have an enabled and
+disabled state.
+
+```js
+class ChocolateDisclaimer extends React.Component {
+  constructor (props) {
+    super(props);
+    this.state = {
+      // a customer first has to promise not to store the chocolate in the
+      // fridge
+      enabled: false
+    };
+  }
+  handleClick () {
+    this.setState({
+      enabled: !this.state.enabled
+    });
+  }
+  render () {
+    return (
+      <form>
+        <label>
+          <input type='checkbox' checked={this.state.enabled} onClick={this.handleClick} />
+          I agree to never, ever store chocolate in the fridge.
+        </label>
+        <button onClick={...} disabled={this.state.enabled}>Submit order</button>
+      </form>
+    )
+  }
+}
+```
+
+As we can see, this component looks fairly similar to the toggle button above.
+The only real difference is that instead of a toggle "button", we now have an
+input of type `checkbox`. Other than that, everything remains the same.
+
+Wouldn't it be nice if we could de-duplicate this logic and extract out the
+redundant `handleClick` handler?
+
+### Modularizing actions
+
+If we look at the code of the toggle button and chocolate disclaimer, we notice
+that the click handlers are completely **identical**.
+
+Code redundancy is never a good thing, therefore we should try to extract out
+our `handleClick` function.
+
+One way would be to have a super-class `Toggleable` that both
+`ChocolateDisclaimer` and `ToggleButton` inherit from:
+
+```js
+class Toggleable extends React.Component {
+  toggleState () {
+    this.setState({
+      enabled: !this.state.enabled
+    });
+  }
+}
+```
+
+`ToggleButton` could then inherit from this superclass:
+
+```js
+class ToggleButton extends Toggleable {
+  constructor (props) {
+    super(props);
+    this.state = {
+      // initially our `ToggleButton` is 
+      enabled: false
+    };
+  }
+  render () {
+    return (
+      <button onClick={this.toggleState}>
+        {this.state.enabled ? 'Enabled' : 'Disabled'}
+      </button>
+    );
+  }
+}
+```
+
+This is already a bit better, but becomes quickly unmanageable. A component
+might for instance be `Toggleable` and `Submittable`… which becomes a bit messy
+very quickly. JavaScript doesn't support multiple-inheritance (inheriting from
+more than one class at once) and building our own mixin system is a bit
+unnecessary.
+
+Instead, we can extract out our handler into a separate "action".
+
+**Advanced:** Redux also uses actions. We're trying to slowly move towards such
```

Comment:
While I like the callout here to let students know what's on the horizon, we've never explicitly talked about Redux and Flux at this point — they'll be unknown concepts to most students. Maybe this might be a little too confusing?